tests/conftest.py

```diff
@@ -1329,3 +1329,13 @@ def admin(UserFixture, app, db, admin_role_need):
     datastore.add_role_to_user(u.user, role)
     db.session.commit()
     return u
+
+
+@pytest.fixture(scope="module")
+def cli_runner(base_app):
```

Comment:
comment/minor: isn't this fixture already available [from `pytest-invenio`](https://github.com/inveniosoftware/pytest-invenio/blob/779fab4eb1ed899ff3445c075705776782920b64/pytest_invenio/fixtures.py#L363)?

Reply:
I think it is, I moved it upper in the hierarchy but I think it can in fact be removed.

Reply:
Actually it's not 😃 

Their signature is slightly different:

- [pytest](https://github.com/inveniosoftware/pytest-invenio/blob/779fab4eb1ed899ff3445c075705776782920b64/pytest_invenio/fixtures.py#L375):
  ```python
  def cli_invoke(command, input=None, *args):
       return base_app.test_cli_runner().invoke(command, args, input=input)
   # Usage
   cli_invoke(command, 'test', '-f')
  ```
  This version does not work because `args` is a tuple ('-f') and `input` is not picked by flask cli runner. In reality this works for commands without any args, as in [app-rdm](https://github.com/inveniosoftware/invenio-app-rdm/blob/f228f9d593118261fde0f9b80dad0d52088e8ad7/tests/fixtures/test_cli.py#L18)

- invenio_rdm_records and [invenio-communities](https://github.com/inveniosoftware/invenio-communities/blob/9de35e431b14f9a558e44721e8db0c2faea1f5ef/tests/conftest.py#L421)
  ```python
  def cli_invoke(command, *args, input=None):
        return base_app.test_cli_runner().invoke(command, args, input=input)
  
  # Usage
  cli_invoke(command, '-f', input='test')
  ```
  This version works because `args` is a tuple : `('-f', 'test')` and [flask test cli](https://click.palletsprojects.com/en/8.1.x/api/#click.testing.CliRunner.invoke) likes it. None that `input` is not used either by the cli runner 😄 

That sneaky difference makes all the difference (pun intended).

TL;DR
I'd argue that this needs to be analysed in another issue so we can do some code cleanup. It shouldn't be too hard but I'd prefer to tackle it afterwards so this task isn't blocked further.

Reply:
Nice one spotting this!

Agree, we can shelve this and address separately, since if we fix this it needs coordination across many modules.

Reply:
Created a separate [issue](https://github.com/inveniosoftware/pytest-invenio/issues/94) to adress it.